index.js

```diff
@@ -1,33 +1,46 @@
 import {spawn} from 'node:child_process';
+import {once} from 'node:events';
 import {lineIterator, combineAsyncIterables} from './utilities.js';
 
 export default function nanoSpawn(command, arguments_ = [], {signal, timeout, nativeOptions} = {}) {
 	const subprocess = spawn(command, arguments_, {...nativeOptions, signal, timeout});
 
-	// eslint-disable-next-line no-async-promise-executor
-	const promise = new Promise(async (resolve, reject) => {
-		try {
-			subprocess.on('close', exitCode => {
-				// TODO: Pass in `stdin` and `stdout` strings here.
-				resolve({
-					exitCode,
-				});
-			});
-
-			subprocess.on('error', error => {
-				reject(error);
-			});
-		} catch (error) {
-			reject(error);
-		}
-	});
+	const promise = getResult(subprocess);
 
 	const stdoutLines = lineIterator(subprocess.stdout);
 	const stderrLines = lineIterator(subprocess.stderr);
-
 	return Object.assign(promise, {
 		[Symbol.asyncIterator]: () => combineAsyncIterables(stdoutLines, stderrLines),
 		stdout: stdoutLines,
 		stderr: stderrLines,
 	});
 }
+
+const getResult = async subprocess => {
+	const result = {};
+	bufferOutput(subprocess, result, 'stdout');
+	bufferOutput(subprocess, result, 'stderr');
+
+	try {
+		const [exitCode, signalName] = await once(subprocess, 'close');
+		return {...getOutput(result), exitCode, signalName};
+	} catch (error) {
+		throw Object.assign(error, getOutput(result));
+	}
+};
+
+const bufferOutput = (subprocess, result, streamName) => {
+	result[streamName] = '';
+	subprocess[streamName].on('data', chunk => {
+		result[streamName] += chunk;
+	});
+};
```

Comment:
We cannot use `stream[Symbol.asyncIterator]` (including through `stdoutLines` or through `node:stream/consumers` `text()`) because it uses `stream.read()` under-the-hood, which means it allows only a single consumer.